lib/ansible/vars/manager.py

```diff
@@ -477,9 +477,11 @@ def _get_magic_variables(self, play, host, task, include_hostvars, include_deleg
 
         if task:
             if task._role:
-                variables['role_name'] = task._role.get_name()
+                variables['role_name'] = task._role.get_name(include_role_fqcn=False)
                 variables['role_path'] = task._role._role_path
                 variables['role_uuid'] = text_type(task._role._uuid)
+                variables['ansible_collection_name'] = task._role._role_collection
+                variables['ansible_role_name'] = task._role.get_name()
```

Comment:
Should `ansible_collection_name` and `ansible_role_name`  be listed in `docs/docsite/rst/reference_appendices/special_variables.rst` perhaps?

Reply:
done